### ISSUE TITLE
[Spell][Hunter] Steady Shot, Aimed Shot

### ISSUE THREAD

**Issue by ta55el** [Atlantiss/NetherwingBugtracker#3259]
**Description**: 
Haste formula for Steady Shot is wrong.

**Current behaviour**: 
Steady Shot is casted too long.
Currently formula is wrong cuz cast time is longer than should.
https://imgur.com/a/S3QgvMX

**Expected behaviour**: 
http://wowwiki.wikia.com/wiki/Steady_Shot?oldid=1317570
Steady Shot Cast Time = 1.5 / 1.15 (quiver) / 1.2 (Serpent's Swiftness) = 1.087
https://www.wowhead.com/forums&topic=36691
Comment from wowhead: My usual cast time without haste but with Serpents Swiftness is ~1.1 sec when imp hawk procs it goes down to about .8 sec. Also it does scale with haste as far as I know.

**Server Revision**: 2395

Steady shot is the main skill that is used for dps.
Same problem for Aimed Shot.

**Comment by Easyfraggy** [comment]
Something is definitely odd here - it seems quiver has no effect when using Serpent Swiftness.

**Comment by ta55el** [comment]
Do u use even math? If quiver would have not effect then cast time of Steady Shot would be 1,25 sec...

**Comment by Zarant** [comment]
Just use cheeky's spreadsheet as a reference, go to the shot rotation tab and see the formula they used for steady shot cast time:
https://web.archive.org/web/20081004051530/http://elitistjerks.com:80/f31/t9816-hunter_spreadsheets_--_development/
I don't believe the EJ neckbeards from 2007-2008 would get that formula wrong.

A small snippet of the spreadsheet calculations using a 3.0 speed weapon, standard BM spec and no haste from gear:
![cs1](https://user-images.githubusercontent.com/26726709/49939619-7bde2000-fec4-11e8-832a-4f2dcd9589b7.png)

Total haste = 1.15*1.20 = 1.38
Steady cast time = 1.5/1.38 = 1.09

**Comment by Zarant** [comment]
I just found why the cast time is longer than expected. Looking through the [cmangos code](https://github.com/cmangos/mangos-tbc/blob/322728efa2e055d653a3e058a388311c1393f2e1/src/game/Spells/SpellMgr.cpp#L147) I found this particular comment:

> Hunter Ranged spells need cast time + 0.5s to reflect tooltips, excluding Auto Shot

Apparently, the core interprets steady shot as having 1s cast time, although the tooltip says it's a 1.5s cast time spell.

Currently, if you consider a 20% haste increase from BM talents and a 15% haste bonus from your quiver, the cast time is calculated in the following way: 1/(1.2*1.15) + 0.5 = 1.224

So what is happening here is that the extra 0.5s cast time is being added after haste calculations and not before.
